diff --git a/arch/arm64/boot/dts/qcom/sm8150-xiaomi-nabu.dts b/arch/arm64/boot/dts/qcom/sm8150-xiaomi-nabu.dts
index 864acb5..430e4e9 100644
--- a/arch/arm64/boot/dts/qcom/sm8150-xiaomi-nabu.dts
+++ b/arch/arm64/boot/dts/qcom/sm8150-xiaomi-nabu.dts
@@ -1050,6 +1050,19 @@
 	vccq-max-microamp = <700000>;
 	vccq2-supply = <&vreg_s4a_1p8>;
 	vccq2-max-microamp = <750000>;
+
+	/delete-property/ resets;
+	/delete-property/ reset-names;
+	freq-table-hz =
+		<37500000 300000000>,
+		<0 0>,
+		<0 0>,
+		<37500000 300000000>,
+		<37500000 300000000>,
+		<0 0>,
+		<0 0>,
+		<0 0>,
+		<0 0>;
 };
 
 &ufs_mem_phy {
