diff --git a/arch/arm64/boot/dts/qcom/sm8150-xiaomi-nabu.dts b/arch/arm64/boot/dts/qcom/sm8150-xiaomi-nabu.dts
--- a/arch/arm64/boot/dts/qcom/sm8150-xiaomi-nabu.dts
+++ b/arch/arm64/boot/dts/qcom/sm8150-xiaomi-nabu.dts
+++ b/arch/arm64/boot/dts/qcom/sm8150-xiaomi-nabu.dts	2026-09-10 06:16:16.745823816 +0530
@@ -793,6 +793,24 @@
 &remoteproc_slpi {
  status = "okay";
  firmware-name = "qcom/sm8150/xiaomi/nabu/slpi_nb.mbn";
+	glink-edge {
+		fastrpc {
+			qcom,protection-domain = "tms/servreg", "msm/slpi/root_pd";
+
+			compute-cb@1 {
+				dma-coherent;
+			};
+
+			compute-cb@2 {
+				dma-coherent;
+			};
+
+			compute-cb@3 {
+				dma-coherent;
+			};
+		};
+	};
+
 };
 
 &sound {
