diff --git a/drivers/media/i2c/ov13b10.c b/drivers/media/i2c/ov13b10.c
index e85c7d3..32f89f6 100644
--- a/drivers/media/i2c/ov13b10.c
+++ b/drivers/media/i2c/ov13b10.c
@@ -16,6 +16,12 @@
 #define OV13B10_REG_VALUE_16BIT		2
 #define OV13B10_REG_VALUE_24BIT		3
 
+static const char * const ov13b10_supply_names[] = {
+	"avdd",
+	"dovdd",
+	"dvdd",
+};
+
 #define OV13B10_REG_MODE_SELECT		0x0100
 #define OV13B10_MODE_STANDBY		0x00
 #define OV13B10_MODE_STREAMING		0x01
@@ -706,7 +712,7 @@ struct ov13b10 {
 	struct v4l2_ctrl_handler ctrl_handler;
 
 	struct clk *img_clk;
-	struct regulator *avdd;
+	struct regulator_bulk_data supplies[ARRAY_SIZE(ov13b10_supply_names)];
 	struct gpio_desc *reset;
 
 	/* V4L2 Controls */
@@ -1196,8 +1202,7 @@ static int ov13b10_power_off(struct device *dev)
 
 	gpiod_set_value_cansleep(ov13b10->reset, 1);
 
-	if (ov13b10->avdd)
-		regulator_disable(ov13b10->avdd);
+	regulator_bulk_disable(ARRAY_SIZE(ov13b10_supply_names), ov13b10->supplies);
 
 	clk_disable_unprepare(ov13b10->img_clk);
 
@@ -1216,13 +1221,11 @@ static int ov13b10_power_on(struct device *dev)
 		return ret;
 	}
 
-	if (ov13b10->avdd) {
-		ret = regulator_enable(ov13b10->avdd);
-		if (ret < 0) {
-			dev_err(dev, "failed to enable avdd: %d", ret);
-			clk_disable_unprepare(ov13b10->img_clk);
-			return ret;
-		}
+	ret = regulator_bulk_enable(ARRAY_SIZE(ov13b10_supply_names), ov13b10->supplies);
+	if (ret < 0) {
+		dev_err(dev, "failed to enable regulators: %d", ret);
+		clk_disable_unprepare(ov13b10->img_clk);
+		return ret;
 	}
 
 	gpiod_set_value_cansleep(ov13b10->reset, 0);
@@ -1490,13 +1493,15 @@ static int ov13b10_get_pm_resources(struct device *dev)
 		return dev_err_probe(dev, PTR_ERR(ov13b->img_clk),
 				     "failed to get imaging clock\n");
 
-	ov13b->avdd = devm_regulator_get_optional(dev, "avdd");
-	if (IS_ERR(ov13b->avdd)) {
-		ret = PTR_ERR(ov13b->avdd);
-		ov13b->avdd = NULL;
-		if (ret != -ENODEV)
-			return dev_err_probe(dev, ret,
-					     "failed to get avdd regulator\n");
+	{
+		int i;
+		for (i = 0; i < ARRAY_SIZE(ov13b10_supply_names); i++)
+			ov13b->supplies[i].supply = ov13b10_supply_names[i];
+
+		ret = devm_regulator_bulk_get(dev, ARRAY_SIZE(ov13b10_supply_names),
+					    ov13b->supplies);
+		if (ret)
+			return dev_err_probe(dev, ret, "failed to get regulators\n");
 	}
 
 	return 0;
