--- a/src/daemon/Display.cpp
+++ b/src/daemon/Display.cpp
@@ -419,7 +419,9 @@ namespace SDDM {
             for(const SessionInfo &s : reply.value()) {
                 if (s.userName == user) {
                     OrgFreedesktopLogin1SessionInterface session(Logind::serviceName(), s.sessionPath.path(), QDBusConnection::systemBus());
-                    if (session.service() == QLatin1String("sddm") && session.state() == QLatin1String("online")) {
+                    if ((session.service() == QLatin1String("sddm")
+                        || session.service() == QLatin1String("sddm-autologin"))
+                            && session.state() == QLatin1String("online")) {
                         m_reuseSessionId = s.sessionId;
                         break;
                     }
